src/main/java/com/linkedin/kmf/KafkaMonitor.java

```diff
@@ -52,11 +66,13 @@ public KafkaMonitor(Map<String, Map> testProps) throws Exception {
         _services.put(name, service);
       }
     }
-
-    _executor = Executors.newSingleThreadScheduledExecutor();
   }
 
-  public void start() {
+  public synchronized void start() {
+    if (_isShutdown) {
```

Comment:
@smccauliff Sure, I actually agree that we should take care of the general case. Then in the most general case, let's assume people can call all three methods in any order concurrently. Then the current approach is not thread safe. I suppose we intend to prevent the body of `stop()` from being executed multiple times by checking `_isShutdown` at its beginning. Yet it can still be executed multiple times if multiple threads call `stop()` concurrently. It seems that we don't want to execute the body of `start()` if user has finished the body of `shutdown()` by checking `istShutdown` at the beginning of `start()`, still it is possible that the body of `start()` can be executed multiple times for similar reason.

Thus it seems that we should do something similar to what `ProduceService()` does by using `AtomicBoolean _running`. The code style would also be more consistent. Does this make sense?